src/server/cluster/outgoing_slot_migration.cc

```diff
@@ -107,18 +107,35 @@ bool OutgoingMigration::ChangeState(MigrationState new_state) {
 }
 
 void OutgoingMigration::Finish(bool is_error) {
-  const auto new_state = is_error ? MigrationState::C_ERROR : MigrationState::C_FINISHED;
-  if (!ChangeState(new_state)) {
-    return;
+  bool should_cancel_flows = false;
+
+  std::lock_guard lk(state_mu_);
```

Comment:
I think it's better to wrap 112-129 into block to avoid locking over `AwaitFiberOnAll`.